opencsp/app/sofast/SofastGUI.py

```diff
@@ -133,6 +111,17 @@ def _plot_image(self, ax: plt.Axes, image: np.ndarray, title: str = '', xlabel:
         ax.set_xlabel(xlabel)
         ax.set_ylabel(ylabel)
 
+    @functools.cached_property
```

Comment:
This is a cool decorator, didn't know that existed!

Reply:
Right?! Pretty nifty. Also extremely susceptible to bugs if not used with care since by definition it produces hidden side effects.